tests/test_permissions_mechanism.py

```diff
@@ -32,55 +38,229 @@ class MainPermission(BaseTestPermission):
         ]
         self.assertEqual(PERMISSIONS_ORDER, expected)
 
+    @override_settings(PERMISSIONS=['tests.base.TestGivingPermission'], TOKEN='secret-token')
     async def test_giving_permission(self):
         """Tests the case when the permission is giving."""
-        screen = TestScreen()
-        source = cast('Handler[..., State]', screen.move)
 
-        permission_instance = TestGivingPermission()
-        handler = permission_instance.check_permission(source)
+        class TestScreen(Screen):
+            """The class implements a screen for this test."""
 
-        state = await handler(self.update, self.context)
+            description = 'A test description.'
+
+        self._init_application([TestScreen])
+        screen = TestScreen()
+
+        state = await screen.move(self.update, self.context)
         self.assertEqual(state, DEFAULT_STATE)
 
+    @override_settings(PERMISSIONS=['tests.base.TestDenyingPermission'], TOKEN='secret-token')
     async def test_denying_permission(self):
         """Tests the case when the permission is denied."""
-        screen = TestScreen()
-        source = cast('Handler[..., State]', screen.move)
 
-        permission_instance = TestDenyingPermission()
-        handler = permission_instance.check_permission(source)
+        class TestScreen(Screen):
+            """The class implements a screen for this test."""
 
-        state = await handler(self.update, self.context)
+            description = 'A test description.'
+
+        self._init_application([TestScreen])
+        screen = TestScreen()
+
+        state = await screen.move(self.update, self.context)
         self.assertEqual(state, PERMISSION_DENIED_STATE)
 
+    @override_settings(PERMISSIONS=[
+        'tests.test_permissions_mechanism.TestPermissionWithSyncChecker',
+    ], TOKEN='secret-token')
     async def test_sync_permission_denied(self):
         """Tests the case when the permission checker is a synchronous."""
-        screen = TestScreen()
-        source = cast('Handler[..., State]', screen.move)
 
-        permission_instance = TestPermissionWithSyncChecker()
-        handler = permission_instance.check_permission(source)
+        class TestScreen(Screen):
+            """The class implements a screen for this test."""
+
+            description = 'A test description.'
 
-        state = await handler(self.update, self.context)
+        self._init_application([TestScreen])
+        screen = TestScreen()
+
+        state = await screen.move(self.update, self.context)
         self.assertEqual(state, PERMISSION_DENIED_STATE)
+
+    @override_settings(PERMISSIONS=['tests.base.TestDenyingPermission'], TOKEN='secret-token')
+    async def test_ignoring_one_permission(self):
+        """Tests the case when one permission is ignored."""
+
+        class ScreenWithIgnorePermissionHandler(Screen):
+            """The class implements a screen which handler ignores
+            TestDenyingPermission permission.
```

Comment:
The class implements a screen with a handler that ignores the TestDenyingPermission permission.